### PR TITLE
Default flipper toggle to off in development

### PR DESCRIPTION
**Pull request by kfrz**
## Description of change

The current behavior is for features to be [enabled by default](https://github.com/department-of-veterans-affairs/vets-api/blob/master/config/initializers/flipper.rb) in development environments.

This is causing problems for developers, because the SSOe flag is defaulted to enabled but SSOe is not ready for developers to use locally. See [this DSVA Slack thread](https://dsva.slack.com/archives/CBU0KDSB1/p1589216158357600).

## Original issue(s)
department-of-veterans-affairs/va.gov-team#8519

## Things to know about this PR

Features are now still enabled by default in development but can be disabled by default in a given environment by adding a key to feature in `config/features.yml` like:

```yaml
feature_name
  default_enabled_environments:
    development: false
```

Features can also be toggled in development by visiting `/flipper`

### CODE REVIEW

File: config/initializers/flipper.rb
Comment (LindseySaari) [inline config/initializers/flipper.rb:49]: Can you update the comment -- "default features to enabled for development"

**Comment by annaswims** [comment]
This looks good to me, but I'd lean towards updating the config to default to "true" for all of the features that are enabled in staging.  That way a FE dev who occasionally starts vets-api sees behavior that's closer to what they'd expect.

**Comment by kfrz** [comment]
@annaswims your comment implies a reversal of the logic from the original approach, now to individually configure each feature to match https://api.va.gov/flipper/features we'll need to add an override key:value to each feature in `config/features.yml`

I don't have the cycles to go through and manually update config to match production right now - we'll either have to merge the code as originally intended or not merge it at all right now. Either way requires a circle back on the ticket.

File: config/initializers/flipper.rb
Comment (annaswims) [inline config/initializers/flipper.rb:44]:
```suggestion
FLIPPER_FEATURE_CONFIG['features'].each do |feature, feature_config|
```
if you change this line you can get rid of line 45

**Comment by annaswims** [comment]
I suspect tests will pass after master is merged and the CVE is ignored.

File: config/initializers/flipper.rb
Comment (erluti) [inline config/initializers/flipper.rb:50]: it looks like this was previously enabling all of the features for test, but with no keys for test in `features.yml`, test will not enable any of these.

Comment (erluti) [reply config/initializers/flipper.rb:50]: Or the key could be more explicit, like `dev_disabled_by_default` and it could be a bit more like the original method:
```
Flipper.enable(feature) if (Rails.env.development? && !feature_config['dev_disabled_by_default']) || Rails.env.test?
```

File: config/features.yml
Comment (mpelzsherman) [inline config/features.yml:20]: We are no longer using `facility_locator_show_community_cares`.

Can you please replace it with this?

```yml
  facilities_ppms_suppress_community_care:
    actor_type: user
    description: Hide ppms community care searches
    default_enabled_environments:
      development: false
```

**Comment by mpelzsherman** [comment]
Any ETA on this? It would save us some time flipping off features every time we do a build!

**Review by kfrz — COMMENTED** [review]
I think this accomplishes the goals - @mpelzsherman @annaswims ready for final review to make sure it meets your expectations

**Review by annaswims — APPROVED** [review]
happy

 YES!

File: config/initializers/flipper.rb
Comment (ericboehs) [inline config/initializers/flipper.rb:48]: It seems we shouldn't be trying to enable a feature unless it exists. Is there a reason to exclude this from `.exists?` conditional?

File: config/features.yml
Comment (ericboehs) [inline config/features.yml:5]: This line sounds like it was written for the commit rather than this file as it looks like it's already set to `true` for all features. Should the copy read `false` here? 😕

File: config/initializers/flipper.rb
Comment (erluti) [reply config/initializers/flipper.rb]: this was an oversight, good catch!

**Review by LindseySaari — APPROVED** [review]
💯

**Comment by erikphansen** [comment]
Why was `enable_in_development: true` set for only some of the existing feature flags? Profile 2.0 stopped showing up locally after this change. If I hadn't seen one mention of this change in one VSA channel I probably would have had no idea why things were behaving strangely locally.

**Pull request merged**
